### PR TITLE
contracts-bedrock: eip 1967 constants in Constants.sol

### LINKED ISSUE #5300: contracts-bedrock: eip 1967 constants in Constants.sol
**Is your feature request related to a problem? Please describe.**
There are multiple places where the EIP 1967 constants are used and they are defined inline multiple times. We should deduplicate the values.

**Describe the solution you'd like**
We should hardcode the constants into `Constants.sol` and then also have unit tests that the values are correct and also import the values from the constants library into all the places where they are used.

### PR DESCRIPTION
**Pull request by fgimenez**
**Description**

Defines eip1967-related constants in Constants.sol and adds changes to use them.

**Tests**

Added unit tests to check the constant values according to https://eips.ethereum.org/EIPS/eip-1967 

**Metadata**

- Fixes #5300

### CODE REVIEW

File: packages/contracts-bedrock/contracts/libraries/Constants.sol [hunk 1]
```diff
@@ -27,6 +27,24 @@ library Constants {
      */
     address internal constant DEFAULT_L2_SENDER = 0x000000000000000000000000000000000000dEaD;
 
+    /**
+     * @notice EIP1967 Logic contract address. Holds the address of the logic contract that a
+     *         proxy delegates to.
+     */
+    bytes32 internal constant PROXY_LOGIC_CONTRACT_ADDRESS = 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc;
```
Comment (tynes) [inline packages/contracts-bedrock/contracts/libraries/Constants.sol:34]: Preference for the word implementation to be used here

File: packages/contracts-bedrock/contracts/libraries/Constants.sol [hunk 2]
```diff
@@ -27,6 +27,24 @@ library Constants {
      */
     address internal constant DEFAULT_L2_SENDER = 0x000000000000000000000000000000000000dEaD;
 
+    /**
+     * @notice EIP1967 Logic contract address. Holds the address of the logic contract that a
+     *         proxy delegates to.
+     */
+    bytes32 internal constant PROXY_IMPLEMENTATION_ADDRESS = 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc;
```
[the remark this answers is not in the document]
Comment (fgimenez) [reply packages/contracts-bedrock/contracts/libraries/Constants.sol]: done, thanks!

**Comment by tynes** [comment]
@fgimenez can you rebase on `develop` + squash your commits?

**Comment by fgimenez** [comment]
> @fgimenez can you rebase on `develop` + squash your commits?

sure, done

**Review by tynes — APPROVED** [review]

**Comment by fgimenez** [comment]
Not sure if relevant but I see some errors in `contracts-bedrock-slither`:
```
Constants.PROXY_IMPLEMENTATION_ADDRESS (contracts/libraries/Constants.sol#35-36) is never used in Constants (contracts/libraries/Constants.sol#12-73)
Constants.PROXY_BEACON_CONTRACT_ADDRESS (contracts/libraries/Constants.sol#43-44) is never used in Constants (contracts/libraries/Constants.sol#12-73)
Constants.PROXY_OWNER_ADDRESS (contracts/libraries/Constants.sol#51-52) is never used in Constants (contracts/libraries/Constants.sol#12-73)
```
I've tried to add `//slither-disable-next-line unused-state-variable` but doesn't seem to work, any ideas?

**Comment by fgimenez** [comment]
> I've tried to add //slither-disable-next-line unused-state-variable but doesn't seem to work, any ideas?

finally added the new slither findings related to constants to `slither.db.json` with `yarn slither:triage`, also removed the unused `PROXY_BEACON_CONTRACT_ADDRESS`

**Comment by fgimenez** [comment]
is there anything missing here? please let me know if I should change anything, thx!

**Comment by tynes** [comment]
> is there anything missing here? please let me know if I should change anything, thx!

Sorry about the slow progress on this, we have been focused on audit fixes. This work is appreciated, and I believe the PR was opened at a time when we broke CI, so if you rebase then CI passes then it is in a good place

**Comment by fgimenez** [comment]
> This PR is stale because it has been open 14 days with no activity. Remove stale label or comment or this will be closed in 5 days.

ping @tynes is there anything I can do to have this merged? I can close it now if it is going to be closed anyway bc of inactivity.

**Comment by fgimenez** [comment]
> I'd like to have this merged but unfortunately we are in a governance mandated code freeze for the contracts that this code touches. We should be able to merge this after mainnet is upgraded on June 6th.

thank you so much for the update, will take care of keeping it open until mainnet upgrade

**Review by refcell — APPROVED** [review]

**Comment by refcell** [comment]
Heads up @fgimenez - opened a pr up preserving your commit history but rebased with our latest changes. https://github.com/ethereum-optimism/optimism/pull/6912

**Comment by fgimenez** [comment]
> Heads up @fgimenez - opened a pr up preserving your commit history but rebased with our latest changes. #6912

awesome thank you so much @refcell! let me know if there's anything else pending on my side

**Comment by refcell** [comment]
Awesome, we were able to get #6912 merged, so will close this out now

By the way, we're working hard to migrate all roadmap and outstanding work to github via issues and project tracking to make our work super transparent. If you ever want to upstream some contributions, feel free to open an issue or even pick up any of the outstanding open issues!!

**Pull request closed without merging**